lib/lamby/rack.rb

```diff
@@ -71,7 +71,8 @@ def query_string
           value.map{ |v| "#{key}=#{v}" }.join('&')
         end.flatten.join('&')
       else
-        event['queryStringParameters'].try(:to_query)
+        query = event['queryStringParameters']
+        query && Rack::Utils.build_query(query)
```

Comment:
This is not going to work. Thoughts on pulling in only the parts of ActiveSupport we need vs all?

```
> q
=> {:brand=>"Gildan", :colors=>["red", "green"]}
> q.to_param
=> "brand=Gildan&colors%5B%5D=red&colors%5B%5D=green"
> Rack::Utils.build_query(q)
=> "brand=Gildan&colors=red&colors=green"
```

Reply:
There is `Rack::Utils.build_nested_query` we may be able to use that should handle array queries.

But yes, we may need to pull in ActiveSupport for what we need.

Reply:
> But yes, we may need to pull in ActiveSupport for what we need.

To that point... one think I remember the core team saying is that the heard the community on ActiveSupport. Since Rails 4.2 (IIRC) they made signifiant efforts in ensuring you can require only what you need. I've seen that too. So memory/hack wise, the footprint is low, but there is disk space for the gem. But that is only ~2MB. But I get it. So I am open to including ActiveSupport code in this gem. But pulling out all the code to make param string work may be tricky.

Again, I appreciate the effort here. Your call on how to move forward. Cheers!

Reply:
So I converted the call to `Rack::Utils.build_nested_query` and we have to manually percent-encode the square brackets because of a known defect in Rack: https://github.com/rack/rack/issues/792.

This should keep the same behaviour as `Hash#to_query` from ActiveSupport.

It would be really awesome if we could avoid pulling in ActiveSupport if at all possible (unless also loading Rails). That way developers could build super-lean rack apps if they wanted to!